docs/provider/static.rst

```diff
@@ -30,33 +31,57 @@ already running resources.
        # the external interface
-       start:   # First available ip
-       end:     # Last available ip
-       cidr:    # cidr notation to describe the pool of ips
-       gateway: # Ip address of the gateway
-       dns:     # Ip address of the DNS
+       start:   203.0.113.17     # First available IP
+       end:     203.0.113.31     # Last available IP
+       cidr:    203.0.113.16/28  # cidr notation to describe the pool of IPs
+       gateway: 203.0.113.254    # IP address of the gateway
+       dns:     203.0.113.253    # IP address of the DNS
      eths:
        - eth1   # Name of the APIs interface
        - eth2   # Name of the external interface
-
    resources: # An object with roles and there associated resources
      control:   # The role control, followed by resource info:
-       address:   # Ip of the control node
-       alias:     # Name for the node (optional)
-       user:      # ssh user name (optional)
-       keyfile:   # path to ssh private key (optional)
-       port:      # ssh port (optional)
-       extra:     # Extra variables passed to ansible (optional)
-         ansible_become: yes
-
+       address:   192.0.2.17  # IP of the control node
+       alias:     control   # Name for the node (optional)
+       user:      user      # ssh user name (optional)
+       keyfile:   keyfile   # path to ssh private key (optional)
+       port:      22        # ssh port (optional)
+       extra:   # Extra variables passed to ansible (optional)
+         ansible_become: yes  # currently mandatory for a successful deploy
      compute:   # A role can also have a list of resources
-       - address: # Ip address of the first compute
-         alias:   # ...
-         user:    # ...
-         keyfile: # ...
-         port:    # ...
-       - address: # Ip address of the second compute
-       - address: # Ip address of the third compute
+       - address: 192.0.2.33  # IP address of the first compute
+         alias:   compute1  # ...
+         user:    user      # ...
+         keyfile: keyfile   # ...
+         port:    22        # ...
+         extra:   # Extra variables passed to ansible (optional)
+           ansible_become: yes  # currently mandatory for a successful deploy
+       - address: 192.0.2.34  # IP address of the second compute
+       - address: 192.0.2.35  # IP address of the third compute
+     network:
+       - address: 192.0.2.17  # IP address
+     storage:
+       - address: 192.0.2.33  # IP address
+       - address: 192.0.2.34  # IP address
+       - address: 192.0.2.35  # IP address
+
+All the hosts must have an IP address already configured on the APIs
+interface matching the the networks defined in ``reservation.yml``
+under ``network``, for example: 198.51.100.17 for the ``control``
+host, 198.51.100.33 for the ``compute1`` host and so on.
+
+In the ``resources`` there must be an host group entry for each of the
+following names:
+
+- control
+- compute
+- network
+- storage
```

Comment:
Is `storage` really required ?

Reply:
It seems so: https://github.com/BeyondTheClouds/enos/blob/master/enos/utils/extra.py#L27

Reply:
Maybe for storage there are exception, I am not sure how to document them. copying the same comment lines for extra.py seems reasonable to me.

Reply:
We can keep only `control`, `compute` and `network`. In `extra.py` we are only saying that if the group isn't specified in the `reservation.yaml` we will generate an empty one in the generated inventory (e.g an inventory without a `storage` group isn't valid whereas an inventory with an empty `storage group is valid`).